### ISSUE TITLE
For iOS devices when pinch zooming the pdf content gets trimmed and does not show after zoom above 100%

### ISSUE THREAD

**Issue by manishagitker** [stephanrauh/ngx-extended-pdf-viewer#2559]
Project is using Angular 16 and Ionic 7, Node 16 . We are opening a pdf under ionic modal.
On zooming in the pdf content get trim and show the white-patch on the pdf. It gets to its normal size only after if zoom out to below 100%

Version of ngx-extended-pdf-viewer : "18.1.9

**Comment by stephanrauh** [comment]
Does it get better if you modify `pdfDefaultOptions.activateWillReadFrequentlyFlag=true` or `=false`? For those readers using version 21+, that's `pdfDefaultOptions.enableHWA`.

**Comment by manishagitker** [comment]
@stephanrauh , This change pdfDefaultOptions.activateWillReadFrequentlyFlag=true/false , didnot fix the issue is still there.

**Comment by stephanrauh** [comment]
I've managed to reproduce the bug on https://mozilla.github.io/pdf.js/web/viewer.html. Can you check this, please? This URL is the showcase of the base library, and my project inherits every bug of the base library. I believe your issue is caused by the base library, so I'd like to ask you to report the bug over there.

**Comment by manishagitker** [comment]
@stephanrauh , we are trying to upgrade to version 21.4.0 , so does this version supports ios v14,15,16 ??
Ios 17 and iOS 18 working fine with above project details mentioned in above description.

**Comment by stephanrauh** [comment]
Good questions. I've seen so many nasty surprises recently I don't want to promise anything. Just search for "Promise.withResolvers" in the issues.

However, if either  https://mozilla.github.io/pdf.js/web/viewer.html or https://mozilla.github.io/pdf.js/legacy/web/viewer.html works on your iOS device, then ngx-extended-pdf-viewer should work, too. It's always possible that there's an error in my branch that's not in the original library, so there are no promises, but the goal is to deliver the same level of compatibility.

**Comment by manishagitker** [comment]
@stephanrauh , I checked the urls https://mozilla.github.io/pdf.js/web/viewer.html  and https://mozilla.github.io/pdf.js/legacy/web/viewer.html  , seems these both are also not supporting on iOS 14 & iOS15. 

But,  is there a plan to support these OS in coming releases , also the Zoom issue which didnot work beyond 100% on iOS ??

**Comment by stephanrauh** [comment]
No, I don't think so. I can't do that myself, both because I'm short of time and because I don't know how the renderer works, and the pdf.js team probably won't do it because pdf.js is the viewer of Firefox. I don't know their update policy, but to me, it seems unlikely they backport a bugfix for an old iOS version to an old version of pdf.js.

However, I've just installed Firefox on iOS. When you open a PDF document in Firefox, pinching and zooming works fine. However, when you open the pdf.js showcase in Firefox on iOS, zooming is broken. Maybe you can report that bug, and if you're lucky, they fix it. Here's the URL of the bug tracker: https://github.com/mozilla/pdf.js/issues

Please don't mention this project when you report the bug. Just tell them you've been able to reproduce the bug in Firefox on iOS 17. Usually, the team only accepts bugs that are relevant to Firefox.

**Comment by stephanrauh** [comment]
I'm closing the bug ticket because it's a bug of the base library, so it has to be fixed there.
